docs/api/guidelines.md

```diff
@@ -0,0 +1,342 @@
+# API Development Guidelines
+
+## Overview
+
+The purpose of this guide is to help you create REST based APIs which are:
+  * simple;
+  * cohesive;
+  * resource based; and
+  * internally consistent
+
+Good APIs can be easily understood by humans, and can be easily implemented
+into other software. Resource based APIs are defined by _endpoints_ (resources)
+which comprise the _nouns_ of the API, and _methods_ (such as GET, PUT, POST,
+etc.) which define the _verbs_.
+
+This guide only covers REST based APIs, however, there are times when it may be
```

Comment:
Should we be mentioning these things here our sticking to a concise description of our API practices?